arwen/contexts/managedType.go

```diff
@@ -56,6 +58,19 @@ func NewManagedTypesContext(host arwen.VMHost) (*managedTypesContext, error) {
 	return context, nil
 }
 
+func initRandomizer(host arwen.VMHost) *math.SeedRandReader {
+	randomizer, _ := math.NewSeedRandReader(append(host.Blockchain().CurrentRandomSeed(), host.Runtime().GetCurrentTxHash()...))
+
+	return randomizer
+}
+
+func (context *managedTypesContext) GetRandReader(ctx unsafe.Pointer) *math.SeedRandReader {
```

Comment:
add comment to public function

Reply:
and use and Interface as return here. and keep the context.Reader as interface as well.